### PR TITLE
Mention eager loading in the guide

### PR DESCRIPTION
**Pull request by kirs**
Fixes https://github.com/rails/rails/issues/29920

Let's add some info about eager loading and `eager_load_paths` into the autoload guide.

### CODE REVIEW

File: guides/source/autoloading_and_reloading_constants.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ This guide documents how constant autoloading and reloading works.
 After reading this guide, you will know:
 
 * Key aspects of Ruby constants
-* What is `autoload_paths`
+* What is `autoload_paths` and `eager_load_paths`
```
Comment (vipulnsward) [inline guides/source/autoloading_and_reloading_constants.md]: *are

File: guides/source/autoloading_and_reloading_constants.md [hunk 2]
```diff
@@ -488,6 +488,48 @@ INFO. `autoload_paths` is computed and cached during the initialization process.
...
+```
+
+But in production, code listed in `autoload_paths` will not be loaded:
+
+```
+$ rails runner -e production "puts Tools"
+uninitialized constant Tools
+```
+
+Use `eager_load_paths` to declare directories to be autoloaded in development and eager loaded in production.
```
Comment (vipulnsward) [inline guides/source/autoloading_and_reloading_constants.md]: this is confusing, saying it does different things in development vs production, can we rephrase it?

**Review by vipulnsward — CHANGES_REQUESTED** [review]

**Comment by rafaelfranca** [comment]
@kirs can you work on @vipulnsward comments?

File: guides/source/autoloading_and_reloading_constants.md [hunk 3]
```diff
@@ -488,6 +488,48 @@ INFO. `autoload_paths` is computed and cached during the initialization process.
 The application needs to be restarted to reflect any changes in the directory
 structure.
 
+eager_load_paths
+----------------
+
+Eager loading is the process of loading all application classes, executed once on application boot
+when cache classes is enabled (e.g. in production).
```
Comment (rafaelfranca) [inline guides/source/autoloading_and_reloading_constants.md]: I think cache classes doesn't change the behavior or eager loading.

**Comment by kamipo** [comment]
Closing this in favor of #31755.

**Pull request closed without merging**